File: web/src/domain/value-objects/cell-shape.ts
import shapes from "@shared/shapes.json" with { type: "json" };

export const CellShape = {
	Rounded: "rounded",
	Square: "square",
	Circle: "circle",
	Dot: "dot",
	Hex: "hex",
} as const;

export type CellShape = (typeof CellShape)[keyof typeof CellShape];

const RENDERABLE_CELL_SHAPES: ReadonlySet<string> = new Set(Object.values(CellShape));

export const CELL_SHAPES: readonly CellShape[] = shapes
	.map((shape) => shape.key)
	.filter((key): key is CellShape => RENDERABLE_CELL_SHAPES.has(key));

export const DEFAULT_CELL_SHAPE: CellShape = CELL_SHAPES[0];

export const isCellShape = (value: string): value is CellShape => (CELL_SHAPES as readonly string[]).includes(value);

File: app/lib/domain/value_objects/cell_shape.dart
enum CellShape {
  square,
  rounded,
  circle,
  dot,
  hex;

  static const CellShape fallback = CellShape.rounded;

  String get label => switch (this) {
    CellShape.square => 'Square',
    CellShape.rounded => 'Rounded',
    CellShape.circle => 'Circle',
    CellShape.dot => 'Dot',
    CellShape.hex => 'Hex',
  };
}
